src/makeRobot.js

```diff
@@ -25,20 +25,105 @@
  *    which will call rescuers and transfer it to the service center
  *    at the coordinates x: 1400, y: 500.
  *
- * @typedef {object} Robot
- * @property {string} name
- * @property {number} wheels
- * @property {number} version
- * @property {function} info
- *
- * @param {string} name
- * @param {number} wheels
- * @param {number} version
- *
- * @return {Robot}
+ * Функція makeRobot створює об'єкт робота
+ * на основі переданих параметрів: рядка name,
+ * числа wheels і version. Робот, що з'являється
+ * з лінії збірки, повинен мати такі можливості:
+
+*
+* @typedef {object} Robot
+* @property {string} name
+* @property {number} wheels
+* @property {number} version
+* @property {function} info
+*
+* @param {string} name
+* @param {number} wheels
+* @param {number} version
+*
+* @return {Robot}
+Надавати інформацію про себе за допомогою геттера info.
+robot.info === 'name:%name%, chip version: %version%, wheels: %wheels%'
+Надавати координати свого розташування за допомогою геттера location.
+robot.location === '%name%: x=14, y=21'
+Мати методи для руху goForward, goBack, goRight та goLeft.
+Методи руху повинні бути використані як ланцюжок.
+robot.goForward().goForward().goForward().goLeft()
+За замовчуванням рухатися на 1 одиницю в правильному напрямку.
+Значення можна збільшити, передаючи потрібне число у метод.
+Від'ємні числа не повинні впливати на розташування робота. goLeft(3)
+Координати робота повинні зберігатись у об'єкті coords з ключами x та y.
+Робот повинен мати можливість запросити
+евакуацію за допомогою robot.evacuate(),
+що викличе службу порятунку і переведе його до сервісного центру
+за координатами x: 1400, y: 500.
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = {
+    name,
+    wheels,
+    version,
+    x: 0,
+    y: 0,
+
+    // #region GETs
+
+    get info() {
+      return (
+        `name: ${this.name}, `
+        + `chip version: ${this.version}, `
+        + `wheels: ${this.wheels}`
+      );
+    },
+
+    get location() {
+      return `${this.name}: x=${this.x}, y=${this.y}`;
+    },
+
+    get coords() {
+      return {
+        x: this.x,
+        y: this.y,
+      };
+    },
+
+    // #endregion
+
+    // #region functions
+
+    evacuate() {
+      this.x = 1400;
+      this.y = 500;
+    },
+
+    goForward(moves = 1) {
+      this.y += moves > 0 ? moves : 0;
+
```

Comment:
No need to update value if condition is falsy. Fix all similar cases
```suggestion
 if (moves > 0) {
   this.y += moves;
 }
```